Assets/CreatureAI/Scripts/Util/SoundDetector.cs

```diff
@@ -0,0 +1,77 @@
+using System.Collections;
+using System.Collections.Generic;
+using TMPro;
+using UnityEngine;
+
+public class SoundDetector : MonoBehaviour
+{
+    private float maxDistance = 30f;
+
+    [SerializeField] bool debugMode = true;
+
+    [SerializeField] LayerMask targetMask;
+
+    LayerMask obstacleMask;
+
+    List<Collider2D> hitTargetList = new List<Collider2D>();
+
+    Vector3? targetPosition;
+
+    public void SetTargetMask(LayerMask targetMask)
+    {
+        this.targetMask = targetMask;
+    }
+
+    public List<Collider2D> Detect()
+    {
+        Collider2D[] targets = FindNearColliders();
+        hitTargetList.Clear();
+
+        foreach(Collider2D target in targets)
+        {
+            float distance = Vector3.Distance(transform.position, target.transform.position);
+            AudioSource source = target.GetComponent<AudioSource>();
+            if (distance > target.GetComponent<AudioSource>().maxDistance && source.isPlaying)
```

Comment:
```suggestion
            AudioSource source = target.GetComponent<AudioSource>();
            if (distance > source.maxDistance && source.isPlaying)
```
source 한번 캐싱했으니까 34줄에서 겟컴퍼넌트 안하고 아까 캐싱한 source 쓰면 좋을 것 같은데 다른 이유가 있나?

Reply:
맞네요 수정했습니다